### ISSUE TITLE
Create full documentation

### ISSUE THREAD

**Issue by vkhorikov** [vkhorikov/CSharpFunctionalExtensions#255]
Full documentation with auto-generation from md files

**Comment by hoshinokanade** [comment]
I would be happy to make some effort if given a boilerplate/ setup ready. There are lots of ways of documenting in the form of markdown and simply not sure which is the one you want.

**Comment by vkhorikov** [comment]
Yeah, I plan to work on this in the next 1 or 2 months. Not sure about the template yet either.

**Comment by jeffward01** [comment]
Hello!

I've been searching for a bit, and am wondering if you had a moment to post that blog post. The only documentation I can find on this class (other than the [code](https://github.com/vkhorikov/CSharpFunctionalExtensions/blob/master/CSharpFunctionalExtensions/ValueObject/EnumValueObject.cs)) is:

* https://github.com/vkhorikov/CSharpFunctionalExtensions/issues/274
* https://github.com/vkhorikov/CSharpFunctionalExtensions/issues/314 

Thanks!

**Comment by vkhorikov** [comment]
> Is there any way I can use an int instead of a string?

Yes, the class allows to specify any type: for the Id (`TId`): https://github.com/vkhorikov/CSharpFunctionalExtensions/blob/master/CSharpFunctionalExtensions/ValueObject/EnumValueObject.cs#L8
